scene/main/scene_tree.cpp

```diff
@@ -1417,7 +1410,22 @@ Error SceneTree::change_scene_to_packed(const Ref<PackedScene> &p_scene) {
 	Node *new_scene = p_scene->instantiate();
 	ERR_FAIL_NULL_V(new_scene, ERR_CANT_CREATE);
 
-	call_deferred(SNAME("_change_scene"), new_scene);
+	// If called again while a change is pending.
+	if (pending_new_scene) {
```

Comment:
Could probably check if it's the same scene being set and just return and do nothing.

Reply:
Or I guess we can't, since it's going to be a new instance? Well nevermind then.

Reply:
This function instantiates the scene `Node` itself. It will always be a different one. An option would be to remember the last `p_scene` argument and in `change_scene_to_file()` remember the `p_path` argument, but that's a bit cumbersome.